zerver/lib/events.py

```diff
@@ -470,6 +471,11 @@ def do_events_register(user_profile, user_client, apply_markdown=True,
     else:
         event_types_set = None
 
+    if user_profile.long_term_idle:
+        add_missing_messages(user_profile)
+        user_profile.long_term_idle = False
+        user_profile.save()
```

Comment:
You should pass `update_fields=["long_term_idle"]` here.  That avoids accidentally overwriting other data in certain race conditions.